http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/LeonardoServiceDbQueries.scala

```diff
@@ -14,13 +14,20 @@ import org.broadinstitute.dsde.workbench.leonardo.http.service.{
 }
 import org.broadinstitute.dsde.workbench.leonardo.model.google.{ClusterName, ClusterStatus, IP, OperationName}
 import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, DataprocInfo}
-import org.broadinstitute.dsde.workbench.leonardo.{ClusterInternalId, LabelMap, RuntimeConfig, ServiceAccountInfo}
 import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
 import org.broadinstitute.dsde.workbench.model.google.{GcsBucketName, GoogleProject}
 
 import scala.concurrent.ExecutionContext
 
 object LeonardoServiceDbQueries {
+
+  type ClusterJoinLabel = Query[(ClusterTable, Rep[Option[LabelTable]]), (ClusterRecord, Option[LabelRecord]), Seq]
+
+  def clusterLabelQuery(baseQuery: Query[ClusterTable, ClusterRecord, Seq]): ClusterJoinLabel =
```

Comment:
The general pattern is to do filtering before joining… hence these `join` methods take a base query which should be pre-filtered before join.

Reply:
I admit this is not an intuitive way to use the DB library… but I have been burned by this.